internal/resources/metal/virtual_circuit/resource.go

```diff
@@ -145,95 +167,104 @@ func Resource() *schema.Resource {
 }
 
 func resourceMetalVirtualCircuitCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	client := meta.(*config.Config).NewMetalClientForSDK(d)
-	vncr := metalv1.VirtualCircuitCreateInput{}
-
-	connId := d.Get("connection_id").(string)
-	portId := d.Get("port_id").(string)
-	projectId := d.Get("project_id").(string)
-	name := d.Get("name").(string)
-
-	if _, ok := d.GetOk("vlan_id"); ok {
-		vncr.VlanVirtualCircuitCreateInput = &metalv1.VlanVirtualCircuitCreateInput{
-			ProjectId:   projectId,
-			Name:        &name,
-			Description: metalv1.PtrString(d.Get("description").(string)),
-			Speed:       metalv1.PtrString(d.Get("speed").(string)),
-			Vnid:        metalv1.PtrString(d.Get("vlan_id").(string)),
-		}
+	if _, ok := d.GetOk("virtual_circuit_id"); ok {
```

Comment:
I think we'll also need some logic in `resourceMetalVirtualCircuitDelete` so that, if `virtual_circuit_id` is specified, it does not attempt to delete the virtual circuit (I think the API will error if you try to delete a shared connection's virtual circuit).